### PR TITLE
HA flows Part 11: Add H&S update operation 

### PR DESCRIPTION
**Pull request by niksv**
Part of #5061

### CODE REVIEW

**Review by artHalMirantis — APPROVED** [review]
from the tests and helpers point of view - everything is ok

**Comment by artHalMirantis** [comment]
from the testing prospective  - its ready to merge
the tests are enough to cover basic needs/cases

**Review by artHalMirantis — APPROVED** [review]

File: src-java/flowhs-topology/flowhs-storm-topology/src/main/java/org/openkilda/wfm/topology/flowhs/fsm/haflow/update/actions/RevertPathsSwapAction.java [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+        String pathName = pathId == null ? NO_PATH : format("the path %s", pathId);
+        stateMachine.saveActionToHistory("Ha-flow was reverted to old paths",
+                format("The ha-flow %s was updated with %s", haFlowId, pathName));
+    }
+
+    private PathId getForwardPathId(HaPathIdsPair haPathIdsPair) {
+        return Optional.ofNullable(haPathIdsPair)
+                .map(HaPathIdsPair::getForward)
+                .map(HaFlowPathIds::getHaPathId)
+                .orElse(null);
```
Comment (dmitrii-beliakov) [inline src-java/flowhs-topology/flowhs-storm-topology/src/main/java/org/openkilda/wfm/topology/flowhs/fsm/haflow/update/actions/RevertPathsSwapAction.java:157]: a minor thing: you unwrap an empty optional here, but later, on 126, check for null. So, actually the logic of optional is needed later, i.e. it makes sense for return an optional from this method.

File: src-java/flowhs-topology/flowhs-storm-topology/src/main/java/org/openkilda/wfm/topology/flowhs/fsm/haflow/update/actions/UpdateFlowStatusAction.java [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+@Slf4j
+public class UpdateFlowStatusAction extends
```
Comment (dmitrii-beliakov) [inline src-java/flowhs-topology/flowhs-storm-topology/src/main/java/org/openkilda/wfm/topology/flowhs/fsm/haflow/update/actions/UpdateFlowStatusAction.java:42]: I'd use HA in all related classes if they are not usable for all flows. Although they are in separate packages they will clutter the search and it's easier. Also you use HA in some classes, I suggest to unify the naming. (this is not blocking for this PR, could be done in another PR)

**Review by dmitrii-beliakov — APPROVED** [review]
👍 amazing job!

File: src-java/flowhs-topology/flowhs-storm-topology/src/main/java/org/openkilda/wfm/topology/flowhs/fsm/haflow/update/actions/UpdateFlowStatusAction.java (hunk 2 above)
Comment (niksv) [reply src-java/flowhs-topology/flowhs-storm-topology/src/main/java/org/openkilda/wfm/topology/flowhs/fsm/haflow/update/actions/UpdateFlowStatusAction.java:42]: You are right. I just forgot to rename classes. will do in next PR

**Pull request merged**